community/kernel/src/test/java/org/neo4j/test/LogTestUtils.java

```diff
@@ -110,42 +109,42 @@ public int getCount()
     }
 
     public static void assertLogContains( FileSystemAbstraction fileSystem, String logPath,
-            LogEntry ... expectedEntries ) throws IOException
+            LogEntry... expectedEntries ) throws IOException
     {
-        ByteBuffer buffer = ByteBuffer.allocateDirect( 9 + MAXGTRIDSIZE + MAXBQUALSIZE * 10 );
-        try ( StoreChannel file = fileSystem.open( new File( logPath ), "r" ) )
+        try ( LogEntryCursor cursor = openLog( fileSystem, new File( logPath ) ) )
         {
-            // Always a header
-            LogHeader logHeader = readLogHeader( buffer, file, true );
+            int entryNo = 0;
+            while ( cursor.next() )
+            {
+                assertThat( "The log contained more entries than we expected!",
+                        entryNo < expectedEntries.length, is( true ) );
 
-            // Read all log entries
-            PhysicalLogVersionedStoreChannel channel =
-                    new PhysicalLogVersionedStoreChannel( file, logHeader.logVersion, logHeader.logFormatVersion );
-            ReadableVersionableLogChannel logChannel = new ReadAheadLogChannel( channel, NO_MORE_CHANNELS, 4096 );
+                LogEntry expectedEntry = expectedEntries[entryNo];
+                assertThat( "Unexpected entry at entry number " + entryNo, cursor.get(), is( expectedEntry ) );
+                entryNo++;
+            }
 
-            // Assert entries are what we expected
-            try ( IOCursor<LogEntry> cursor = new LogEntryCursor( logChannel ) )
+            if ( entryNo < expectedEntries.length )
             {
-                int entryNo=0;
-                while (cursor.next())
-                {
-                    assertThat( "The log contained more entries than we expected!",
-                            entryNo < expectedEntries.length, is( true ) );
-
-                    LogEntry expectedEntry = expectedEntries[entryNo];
-                    assertThat( "Unexpected entry at entry number " + entryNo, cursor.get(), is( expectedEntry ) );
-                    entryNo++;
-                }
-
-                if(entryNo < expectedEntries.length)
-                {
-                    fail( "Log ended prematurely. Expected to find '" + expectedEntries[entryNo].toString() +
-                            "' as log entry number " + entryNo + ", instead there were no more log entries." );
-                }
+                fail( "Log ended prematurely. Expected to find '" + expectedEntries[entryNo].toString() +
+                      "' as log entry number " + entryNo + ", instead there were no more log entries." );
             }
         }
     }
 
+    public static LogEntryCursor openLog( FileSystemAbstraction fs, File log ) throws IOException
```

Comment:
This method now is much more optimistic in terms of resource closing then it was before.
Now we will always assume that we can build cursor without exception?
Why do we need  `Objects.requireNonNull` in that case then?